### PR TITLE
{AKS} Add outbound type none and block

### PR DESCRIPTION
**Pull request by MartinForReal**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -258,6 +258,53 @@ def test_aks_create_and_update_with_managed_nat_gateway_outbound(
...
+        )
+
+        create_cmd = (
+            "aks create --resource-group={resource_group} --name={name} "
+            "--vm-set-type VirtualMachineScaleSets -c 1 "
+            "--aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/NetworkIsolatedClusterPreview "
+            "--outbound-type block "
+            "--bootstrap-artifact-source Cache "
+            "-k 1.30 "
+            "--enable-apiserver-vnet-integration "
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:284]: [Live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=103203892&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=a26fe913-e2be-5062-b08e-d15f1acc7ea2&l=2737) failed with the following error
```
E                   azure.cli.core.azclierror.BadRequestError: (BadRequest) Feature Microsoft.ContainerService/EnableAPIServerVnetIntegrationPreview is not enabled. Please see https://aka.ms/aks/previews for how to enable features.
E                   Code: BadRequest
E                   Message: Feature Microsoft.ContainerService/EnableAPIServerVnetIntegrationPreview is not enabled. Please see https://aka.ms/aks/previews for how to enable features.
```

File: src/aks-preview/HISTORY.rst [hunk 2]
```diff
@@ -13,6 +13,10 @@ Pending
 +++++++
 * Vendor new SDK and bump API version to 2024-07-02-preview.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:14]: Please include this line in your newly added version.

Comment (MartinForReal) [reply src/aks-preview/HISTORY.rst]: updated.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 1 above)
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:284]: [`live test build`](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=103209854&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=b162b355-d59d-5864-ce0f-0a70f12dd28b) succeeded in staging env.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**